### DISCUSSION TITLE
Refonte du système de fight

### DISCUSSION THREAD

**Discussion by Thesmokingllama** [Crownicles/Crownicles#1305] [Idées acceptées / déjà implémentées]
Je me permets d'ouvrir cette discussion, elle pourra potentiellement être utilisée pour toute proposition de changement concernant le nouveau système de fight.
   Je vais donc commencer les hostilités avec ma proposition de nouveau système, parce que je pense que celui-ci n'a pas suffisamment de potentiel et qu'il serait plus sage de repartir sur une base nouvelle plutôt que chercher vainement à l'améliorer. Mais je tiens à dire que la diversification des attaques en fonction de la classe est une feature vraiment top à laquelle je tiens énormément donc l'ensemble de mon propos ne concerne que la façon dont sont calculées les stats en combat. Voici donc mon idée de comment fonctionnerait ce système de fight moins imparfait : 

- Pour commencer on mettrait fin à ce pourcentage de puissance sur les attaques pour le remplacer par des calculs qui interviennent sur les stats des joueurs pendant les combats, de cette manière on conserve l'idée que la progression des stats se fait de moins en moins importante au fil de leur progression (vu que l'ancien système favorisait la différence de stats importantes, par exemple 400-450 à une différence plus faible de 200-250) ce qui a pour conséquence d'aplatir la différence de niveau entre les joueurs durant les fight comme souhaité dans l'esprit de cette mise à jour tout en redonnant de la cohérence dans la puissance de chacun (et ne plus voir des combats où avoir 250 d'attaque est pareil que 600).

- Ce qui amène à mon deuxième point car ici on se dira probablement "oui mais du coup les haut level pourront nous oneshot" et ceux qui se diraient ça auraient raison, en effet, ce n'est pas suffisant de proposer le remplacement du pourcentage de puissance sur les attaques par des calculs, il nous faudrait conserver un système de cap des attaques mais bieeeen plus soft qui permettrait de voir des différences de stats dans leur efficacité en combat (il me semble logique qu'un paladin fasse moins qu'un canonnier mais résiste mieux, que ce soit en attaque ou en défense on a l'impression que tout le monde à la même puissance et la même résistance).

**Comment by Pietagorh** [comment]
Je serais plutôt d'accord, surtout avec un système de matchmaking en plus

Disons que les stats sont pas assez récompensées, et qu'entre 2 personnes de même level c'est assez aléatoire qui gagne, même avec 10/15 levels d'écart

**Comment by Pietagorh** [comment]
J'suis en train de penser (pour une attaque on ne peut plus simpliste) à ça

Ancien système :
attaque - défense

Système actuel :
Chépa comment ça fait

Nouveau système :
(sqrt(attaque) - sqrt(défense)) \** 2
Comme ça la différence reste présente mais moins grande

Faudrait qu'je calcule, là j'suis fatigué mais pense qu'en plus on pourrait tweak l'importance des différences avec genre
(sqrt(attaque / x) - sqrt(défense / x)) \** 2 \* x
Avec pour x une valeur fixée à l'avance pour tweak

**Reply by BastLast** [reply]
Jme permet de m'incruster pour dire qu'on peut faire 99% de ce que décrit hito  sans rien coder de nouveau, juste en changeant des valeurs sur le système actuellement en place. Ca serait top de lire le doc avant de proposer des fonctions random xD

**Comment by Gabrielre14** [comment]
Je me permet une intervention, qui n'est pas en rapport avec le calcul des dégâts, mais avec la trop grande proportion de chance. 
Je pense notamment à l'attaque confuse du canonnier. J'entends beaucoup de joueurs se plaindre de cette attaque cancer, mais elle est efficace avec beaucoup de chance. 
Par exemple, cette attaque peut autant m'empêcher de jouer pendant 5 TOURS (dont un où je suis blessé) rien qu'avec une (ou deux je sais plus) attaques confuses. C'était contre HaDeath. Alors qu'un autre moment, on peut lancer une confuse... et ça ne change rien, puisque que l'adversaire peut jouer même en étant confus, ça arrive régulièrement. 
Je suis d'accord pour mettre un peu de random dans les fights, je ne suis pas contre les coups critiques par exemple, mais là c'est vraiment abusé. Une attaque peut être extrêmement cheat (même la meilleure de toutes), ou complètement inutile, rien qu'avec de la chance. 
En remplacement, je propose, afin d'éviter de rendre cette attaque cancer tout en permettant aux canonnier de posséder une attaque "stratégique" autre que taper et taper (ce que font les autres attaques), que l'attaque confuse soit limité à 2 tours maximum. Le 1er tour, l'attaquant est sûr à 100% de faire faire à l'adversaire une action aléatoire ou le blesser. Pour le 2e tour, c'est 1/2 (+ ou -, à varier pour l'équilibrage). Il n'y a pas de 3e tour de confusion, car ça se repose trop sur la chance et ça rend le jeu cancer.

Mais pour éviter de spam l'attaque confuse, car le 1er tour permet de faire faire une action aléatoire à 100%, je propose qu'au-delà de 2 utilisations, chaque nouvelle utilisation de cette attaque baisse de 25% la chance de rendre confus, jusqu'à 0.
Ainsi, cette attaque permet au canonnier un soutien stratégique sûr (pour empêcher une attaque ultime, ou empêcher l'adversaire de mettre une attaque pertinente qui pourrait faire mal), et non aléatoire, (avec toujours une part de chance, vu que la possibilité de blesser l'adversaire existe) tout en évitant de rendre cette attaque vraiment cancer dans certains cas. 
En échange de ce nerf, une augmentation des dégâts des autres attaques (surtout l'attaque canon, qui est risquée et fait peu de dégâts) pourrait être bien.

**Reply by Thesmokingllama** [reply]
sujet évoqué dans une autre issue

**Reply by Gabrielre14** [reply]
ah mince
J'ai vu refonte du système
je me suis dit que ça passe ici... je copie, je supp et je colle là-bas ?

**Comment by romain22222** [comment]
autre suggestion de modification : mettre un hard cap au nombre de tours de l'altération de confusion (genre 3 ou 4 tours)

ca éviterait les combats absurdes dans lesquels une des personnes est confus durant 10 tours de suite

**Reply by BastLast** [reply]
Elle va être grave rework cette attaque je la déteste

**Comment by BastLast** [comment]
je passe cette suggestion en suggestion acceptée suite à #1695
